xi-an-jiaotong-university.csl

```diff
@@ -0,0 +1,560 @@
+      <text macro="publishing"/>
+      <text macro="url-doi"/>
+    </group>
+  </macro>
+  <!-- zh连续出版物 -->
+  <macro name="periodical-layout-zh">
+    <group delimiter=". ">
+      <text macro="author-zh"/>
+      <text macro="title"/>
+      <text macro="year-volume-issue"/>
+      <text macro="publishing"/>
+      <text macro="url-doi"/>
+    </group>
+  </macro>
+  <!-- zh连续出版物中的析出文献 -->
+  <macro name="article-in-periodical-layout-zh">
+    <group delimiter=". ">
+      <text macro="author-zh"/>
+      <text macro="title"/>
+      <text macro="container-periodical"/>
+      <text macro="url-doi"/>
+    </group>
+  </macro>
+  <!-- zh专利文献 -->
+  <macro name="patent-layout-zh">
+    <group delimiter=". ">
+      <text macro="author-zh"/>
+      <text macro="title"/>
+      <group>
+        <text macro="issued-date"/>
+        <text macro="accessed-date"/>
+      </group>
+      <text macro="url-doi"/>
+    </group>
+  </macro>
+  <!-- en参考文献表格式 -->
+  <macro name="entry-layout">
+    <choose>
+      <if type="article-journal article-magazine article-newspaper" match="any">
+        <text macro="article-in-periodical-layout"/>
+      </if>
+      <else-if type="periodical">
+        <text macro="periodical-layout"/>
+      </else-if>
+      <else-if type="patent">
+        <text macro="patent-layout"/>
+      </else-if>
+      <else-if type="chapter paper-conference" variable="container-title" match="any">
+        <text macro="chapter-in-book-layout"/>
+      </else-if>
+      <else>
+        <text macro="monograph-layout"/>
+      </else>
+    </choose>
+  </macro>
+  <!-- zh参考文献表格式 -->
+  <macro name="entry-layout-zh">
+    <choose>
+      <if type="article-journal article-magazine article-newspaper" match="any">
+        <text macro="article-in-periodical-layout-zh"/>
+      </if>
+      <else-if type="periodical">
+        <text macro="periodical-layout-zh"/>
+      </else-if>
+      <else-if type="patent">
+        <text macro="patent-layout-zh"/>
+      </else-if>
+      <else-if type="chapter paper-conference" variable="container-title" match="any">
+        <text macro="chapter-in-book-layout-zh"/>
+      </else-if>
+      <else>
+        <text macro="monograph-layout-zh"/>
+      </else>
+    </choose>
+  </macro>
+  <citation collapse="citation-number" after-collapse-delimiter=",">
+    <sort>
+      <key variable="citation-number"/>
+    </sort>
+    <layout vertical-align="sup" delimiter="," prefix="[" suffix="]">
+      <text variable="citation-number"/>
+    </layout>
+  </citation>
+  <bibliography entry-spacing="0" et-al-min="4" et-al-use-first="3" second-field-align="flush">
+    <!-- zh格式 -->
+    <layout suffix="." locale="zh-CN">
+      <text variable="citation-number" prefix="[" suffix="]"/>
+      <text macro="entry-layout-zh"/>
+    </layout>
+    <!-- en格式 -->
+    <layout suffix=".">
+      <text variable="citation-number" prefix="[" suffix="]"/>
+      <text macro="entry-layout"/>
+    </layout>
+  </bibliography>
```

Comment:
It's necessary to differentiate the chinese and english in this thesis, but this CSL-M (cs:layout) can't be validated.